Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift

```diff
@@ -0,0 +1,379 @@
+            return UIImage(named: "paypal-logo-1", in: Bundle.primerResources, compatibleWith: nil)
+        case .primerTestSofort:
+            return UIImage(named: "sofort-logo", in: Bundle.primerResources, compatibleWith: nil)
+        default:
+            return buttonImage
+        }
+    }()
+    
+    lazy var submitButton: PrimerButton = {
+        let submitButton = PrimerButton()
+        submitButton.translatesAutoresizingMaskIntoConstraints = false
+        submitButton.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        submitButton.isAccessibilityElement = true
+        submitButton.accessibilityIdentifier = "submit_btn"
+        submitButton.isEnabled = false
+        submitButton.setTitleColor(theme.mainButton.text.color, for: .normal)
+        submitButton.backgroundColor = theme.mainButton.color(for: .disabled)
+        submitButton.layer.cornerRadius = 4
+        submitButton.clipsToBounds = true
+        submitButton.addTarget(self, action: #selector(payButtonTapped(_:)), for: .touchUpInside)
+        return submitButton
+    }()
+    
+    // MARK: - Deinit
+    
+    deinit {
+        log(logLevel: .debug, message: "🧨 deinit: \(self) \(Unmanaged.passUnretained(self).toOpaque())")
+    }
+    
+    // MARK: - Overrides
+    
+    override var originalImage: UIImage? {
+        get {
+            _originalImage
+        }
+        set {
+            _originalImage = newValue
+        }
+    }
+    
+    override func validate() throws {
+        guard let decodedClientToken = ClientTokenService.decodedClientToken, decodedClientToken.isValid else {
+            let err = PrimerError.invalidClientToken(userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"], diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            throw err
+        }
+    }
+    
+    override func start() {
+        
+        self.didStartTokenization = {
+            self.submitButton.startAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = false
+        }
+        
+        self.didFinishTokenization = { err in
+            self.submitButton.stopAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = true
+        }
+        
+        self.didStartPayment = {
+            self.submitButton.startAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = false
+        }
+        
+        self.didFinishPayment = { err in
+            self.submitButton.stopAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = true
+            self.willDismissPaymentMethodUI?()
+            self.didDismissPaymentMethodUI?()
+        }
+        
+        super.start()
+    }
+    
+    override func startTokenizationFlow() -> Promise<PrimerPaymentMethodTokenData> {
+        
+        DispatchQueue.main.async {
+            UIApplication.shared.endIgnoringInteractionEvents()
```

Comment:
You probably want to start ignoring here, right? Also I recently made a change and I always call this in `ensure`.